### ISSUE TITLE
[Platform] Add ability for an Admin to abort a Task

### ISSUE THREAD

**Issue by rao-vasireddy** [yugabyte/yugabyte-db#4568]
In the Tasks page, please add an Actions dropdown for the tasks with the option to abort a task. Once the task is aborted it can be shown with a new status of aborted. YW metadata should be reverted to the prior state . This needs further discussion to lock down the requirements.

Aha! Link: https://yugabyte-test.aha.io/features/PLATFORM-402

**Comment by kriteshranjan** [comment]
AI validation notes (`yb-issue-4568-analysis.md`):

# Analysis: yugabyte/yugabyte-db#4568 (Aha PLATFORM-402; delivered under PLAT-2152 / PLAT-2535 / PLAT-2562 / PLAT-2572)

Issue: "[Platform] Add ability for an Admin to abort a Task" — filed 2020-05-22, kind/enhancement,
area/platform + area/ui, priority/high, OPEN (GitHub updated 2020-08-11), zero comments.
Ask: an Actions control on the Tasks page to abort a task; aborted tasks shown with a new
"aborted" status; "YW metadata should be reverted to the prior state" (reporter flagged this
clause as "needs further discussion").

Kind: feature request → behavior-delta validation (current capability vs requested), not a crash
repro. Component: `managed/` (YugabyteDB Anywhere). Local checkout for line citations:
`46a9f7adfd226d7bc3315c38b660eef69bb11196` (master, 2026-08-28); per-tag verification at the
exact resolved tag SHAs listed under "Versions tested".

## Verdict
NOT-REPRODUCED — the requested capability is present on every required target: the task-abort
REST API, `@Abortable` task framework support, the terminal `Aborted` task state, and both the
Tasks-page "Abort Task" action and the task-drawer Abort button exist at all five resolved tag
SHAs (2024.2 / 2025.1 / 2025.2 / 2026.1 / master-preview). Delivered 2021-12/2022-01 (first
preview v2.11.2.0, first stable v2.12.1.0); the issue was never closed because none of the
fulfilling commits referenced #4568. Quick verdict: Fixed. The optional "revert YW metadata"
clause is delivered separately as the task Rollback API on 2025.1+ (see Root cause §3).

## Root cause
Not a defect; a delivered feature gap. Mechanism on master (`46a9f7adfd`):

1. **API + controller.** `managed/src/main/resources/v1.routes:473` maps
   `POST /customers/:cUUID/tasks/:tUUID/abort` to
   `managed/src/main/java/com/yugabyte/yw/controllers/CustomerTaskController.java:366-378`
   (`abortTask`: RBAC `UNIVERSE`/`UPDATE` at `:354-365`, ownership check `:369`, delegates to
   `commissioner.abortTask(taskUUID, false)` `:370`, audits `Audit.ActionType.Abort` `:374-376`
   (`managed/src/main/java/com/yugabyte/yw/models/Audit.java:243-244`), returns
   "Task is being aborted." / "Task is not running." `:372,:377`).
2. **Commissioner + executor.** `managed/src/main/java/com/yugabyte/yw/commissioner/Commissioner.java:277-303`
   `abortTask`: dequeues a still-queued task `:279-282`, rejects non-`@Abortable` types `:283-286`,
   requires `TaskInfo.State.Running` `:287-290`, releases a pause gate `:291-295`, calls
   `taskExecutor.abort(taskUUID, force)` `:296`, moves in-flight backups to `Stopping` `:298-301`.
   `managed/src/main/java/com/yugabyte/yw/commissioner/TaskExecutor.java:495-512` `abort`:
   `runnableTask.abort(Duration.ZERO)` `:507` sets the abort instant (`:1565-1576`) and CASes
   `{Initializing,Created,Running} → State.Abort` `:509-510`. The running task observes the abort
   at the next checkpoint: `:1067-1068` (before `run()`), subtask scheduling/wait
   `:797-827`, `:1538` (`waitFor` throws "is aborted while waiting"); the resulting
   `CancellationException` is persisted as the terminal `TaskInfo.State.Aborted` at `:1088-1089`.
   Abortability is a class-level annotation `@Abortable`
   (`managed/src/main/java/com/yugabyte/yw/commissioner/ITask.java:33-38`, resolved by
   `TaskExecutor.java:243-247`); 67 task classes carry it at v2026.1.1.1 (39 at v2024.2.11.0).
   Platform shutdown force-aborts pending tasks to `Aborted` with `CancellationException("Platform shutdown")`
   (`managed/src/main/java/com/yugabyte/yw/commissioner/TaskQueue.java:207`).
3. **"Aborted" status + UI.** `managed/src/main/java/com/yugabyte/yw/models/TaskInfo.java:111-115`
   defines `Abort(5)` (in progress) and `Aborted(6)` (terminal; in `COMPLETED_STATES` `:78`,
   `ERROR_STATES` `:84`). `Commissioner.java:411` exposes `abortable` in the task-list payload
   (`isTaskAbortable` `:478-487`); `managed/src/main/java/com/yugabyte/yw/models/CustomerTask.java:592-593`
   renders "Aborting task" / "Task aborted". Tasks page:
   `managed/ui/src/components/tasks/TaskList/TaskListTable.jsx:76-91` shows an RBAC-gated
   "Abort Task" action for `Running`/`Created` rows with `abortable`, confirmed via modal `:202`,
   posting through `managed/ui/src/actions/tasks.js:104-105`; `Aborted` rows get "See Details" `:65`.
   Redesigned task drawer: `managed/ui/src/redesign/features/tasks/components/drawerComp/TaskDetailActions.tsx:74-86,145-160`
   (Abort button + `AbortConfirmModal`). RBAC endpoint map: `managed/ui/src/redesign/features/rbac/ApiAndUserPermMapping.ts:533`.
   Delivered as an inline action + drawer button rather than the literal "Actions dropdown".
4. **"Revert YW metadata" clause.** Abort itself does not revert universe metadata: the universe
   lock is released with `updateInProgress=false` and the error string set
   (`managed/src/main/java/com/yugabyte/yw/commissioner/tasks/UniverseTaskBase.java:1448-1469`),
   while `updateSucceeded=false` set at lock time (`:1094`) leaves the universe flagged until a
   Retry (`v1.routes` → `CustomerTaskController.java:290-310`) or Rollback
   (`v1.routes:475` `POST /customers/:cUUID/tasks/:tUUID/rollback` →
   `CustomerTaskController.java:328-348`; `@CanRollback` `ITask.java:26-31`; `canRollback` flag
   `Commissioner.java:436-444`; drawer button `TaskDetailActions.tsx:88-100,126-144`). Rollback
   is opt-in per task class (EditUniverse, RollbackEditUniverse, SoftwareUpgradeYB,
   SoftwareKubernetesUpgradeYB, SwitchoverDrConfig, SwitchoverDrConfigRollback at master) and
   edit-universe rollback is gated by `yb.task.allow_edit_universe_rollback`
   (`managed/src/main/resources/reference.conf:646-647`). YBA also auto-retries tasks aborted by
   its own restart (`reference.conf:611`).

Affected targets: none. Every required target ships the abort capability; no master fix PR is
warranted.

## Trigger matrix
Behavior-delta cells, predicted from source before verification and then observed per tag
(evidence: `/tmp/claude/yb4568/yb4568-tag-verification.txt`, checks C1–C17).

| condition (at resolved tag) | predicted | observed |
|---|---|---|
| `POST /customers/:cUUID/tasks/:tUUID/abort` route → `CustomerTaskController.abortTask` (C1–C3) | present on all 5 | PASS ×5 |
| `Commissioner.abortTask` + `abortable` flag in task payload (C4–C5) | present on all 5 | PASS ×5 (2024.2 uses `isTaskAbortable(taskInfo.getTaskType())`, same mechanism) |
| `TaskExecutor.abort` + terminal `State.Aborted` on `CancellationException` (C6–C7) | present on all 5 | PASS ×5 |
| `@Abortable` annotation, `TaskInfo.State.Abort/Aborted`, "Task aborted" text (C8–C11) | present on all 5 | PASS ×5 |
| Tasks page "Abort Task" action gated on `row.abortable` (C12–C13) | present on all 5 | PASS ×5 (`TaskListTable.js` before 2026.1, `.jsx` at 2026.1) |
| UI action posts to `/tasks/<uuid>/abort`; drawer Abort button (C14–C15) | present on all 5 | PASS ×5 |
| Task Rollback route + `@CanRollback` (revert-metadata half) (C16–C17) | absent on 2024.2, present 2025.1+ | FAIL on 2024.2 only; PASS on 2025.1/2025.2/2026.1/master |
| Control: yugabytedb/yugabyte DB image contains any YBA/yugaware component | absent on all 5 | `NO_YBA` ×5 (`/tmp/claude/yb4568/yb4568-image-check.txt`) |

## Reproduction
No faithful executable reproduction exists with the official `yugabytedb/yugabyte` images: the
deliverable is the YugabyteDB Anywhere control plane (`managed/`), which is a separate product
(`yugabytedb/yugaware` image + PostgreSQL + Prometheus) that DB images never contain. A live YBA
deployment is out of scope for this validation. The matrix is therefore grounded in (a) source
verification of the mechanism at the exact resolved tag SHAs and (b) an observed control that each
resolved DB image ships DB daemons whose `--version` revision equals the verified tag SHA and no
YBA component.

Commands (all read-only; scripts stored as `.txt`). (a) per-tag mechanism verification: `git grep`
at the 4 tags present in the local shallow checkout, raw.githubusercontent.com fetch of the same
paths at the exact SHA for v2025.2.6.0; expected `SUMMARY pass=17 fail=0` for
2025.1/2025.2/2026.1/master and `pass=15 fail=2` for 2024.2 (only C16/C17 rollback). (b) image
control: pulled digest == resolved digest, `yb-master --version` revision == tag SHA, `NO_YBA`.
(c) freshness evidence for the resolved targets (Docker Hub v2 API).

```sh
bash /tmp/claude/yb4568/yb4568-verify-tags-sh.txt /Users/brandonmcnama/src/github.com/yugabyte/yugabyte-db \
  > /tmp/claude/yb4568/yb4568-tag-verification.txt
bash /tmp/claude/yb4568/yb4568-image-check-sh.txt > /tmp/claude/yb4568/yb4568-image-check.txt
python3 /tmp/claude/yb4568/yb4568-dockerhub-query.py > /tmp/claude/yb4568/yb4568-dockerhub-tags.txt
```

Manual check against a live YBA (any 2.12.1+ / 2024.2+ install), for a maintainer with one:
`POST /api/v1/customers/<cUUID>/tasks/<tUUID>/abort` on a running `@Abortable` task (e.g.
EditUniverse) returns `{"success":true,"message":"Task is being aborted."}`; the task then appears
with status `Aborted` in `GET /api/v1/customers/<cUUID>/tasks`, and the Tasks page shows
"Abort Task" for running abortable rows. Not executed here (no YBA instance).

## Versions tested
Resolved at: 2026-09-05T19:07Z
Resolution evidence (re-verified 2026-09-05T19:19Z; raw output in
`/tmp/claude/yb4568/yb4568-dockerhub-tags.txt` and `/tmp/claude/yb4568/yb4568-tag-commit-resolution.txt`):
Docker Hub `tags?name=<branch>&ordering=last_updated` — 2024.2 newest 2024.2.11.0-b36 pushed
2026-08-25T16:02:12Z (prior 2024.2.10.0-b62); 2025.1 newest 2025.1.4.0-b103 pushed
2026-04-20T15:59:42Z (prior 2025.1.3.2-b1); 2025.2 newest 2025.2.6.0-b111 pushed
2026-09-04T22:23:02Z (prior 2025.2.5.2-b5); 2026.1 newest 2026.1.1.1-b2 pushed 2026-08-18T20:03:16Z
(prior 2026.1.1.0-b91; `latest` alias manifest sha256:3926eedf0ff4b99426fbff533128ff16635ed5697a5cf086dde4fafec28e6f75
= 2026.1.1.1-b2); preview `name=2.25.` newest 2.25.2.0-b359 pushed 2025-05-20T18:34:42Z;
`name=2.27.`, `2.29.`, `2.31.`, `2026.2` return count 0 (no newer preview line). Tag→commit via
`github:get_commit`: v2024.2.11.0 = 04544fd94fb4e50d05994784f444b075a4a5dfdb (`[BACKPORT 2024.2][DEVOPS-3852] ...`),
v2025.1.4.0 = 690ac4141e57d060fd834312f396643e8c41f260 (`[BACKPORT 2025.1][#30772] ...`),
v2025.2.6.0 = ae22ef35ff65de63303467d95ac3e6d2382feba9 (`[BACKPORT 2025.2.6][#33558] ...`),
v2026.1.1.1 = 3b010e021f6027744bf9e016b17e108a2eb24e44 (`[BACKPORT 2026.1.1][PLAT-21546] ...`),
v2.25.2.0 = 03bbc9f31eae5c3fecabd8f0afa5e99211e6080d; master HEAD b8477c4284908e7ff9314fb2b31014d12f20cc0c
(2026-09-04T23:18:07Z). In-image `yb-master --version` revision equals the tag SHA for all five
(`/tmp/claude/yb4568/yb4568-image-check.txt`).

aarch64 images: 2024.2.11.0-b36-aarch64 sha256:53c01c56315e245cc901cbc46e93f30fedb1f87833c7b1aa09d20fbeda27a288
(amd64 sha256:6bb00eeb3dd97b93f36716080d2cfe96d1e61f864201d9a36ae6e3aeacb6d9ce);
2025.1.4.0-b103-aarch64 sha256:08f1be0d6e074e0286e70067c12933a8cf536876d50fc43c52b1127d447ecdc0
(amd64 sha256:fa5555536e8fe7d9522fb6c9c55f3c6062549f318b79c5902726097549263771);
2025.2.6.0-b111-aarch64 sha256:4a1f7a94d3b70c61ecaf5e049521d691b16c9bf204ed01bdb6ab75d5beeb623e
(amd64 sha256:5042b5090b49ef49959a4db9d05c90f5da6482aa72265d6ab4a23f8dc9ec0297);
2026.1.1.1-b2-aarch64 sha256:d27eede72bbc669e348e9920318589b4bb68e98204a241ec6bfaba04bf4f6699
(amd64 sha256:36741c614b0a28498a3447a01b952240905a0ad87c9f68dbcbb20e65bc3d2e9f);
2.25.2.0-b359-aarch64 sha256:3f9cd1b8f520cc78e82ace51174be73343a551148bc6c9a9f821a22f1d2858c5
(amd64 sha256:e69ddcd248db7ee5c097d8c955d3657cf7fd9c21743a4589281767487ab5832b).

| target | latest resolved version | source revision | observed repro/control | effective status |
|---|---|---|---|---|
| 2024.2 | 2024.2.11.0-b36 (tag v2024.2.11.0) | 04544fd94fb4e50d05994784f444b075a4a5dfdb | Runtime repro not applicable: DB image has no YBA (`NO_YBA`, digest OK, in-image revision = tag SHA). Source at tag: abort route `v1.routes:460`, `CustomerTaskController.abortTask` `:394`, `Commissioner.abortTask` `:170`, `TaskExecutor.abort` `:452`, `@Abortable` `ITask.java:26`, `State.Aborted` `TaskInfo.java:102`, UI `TaskListTable.js:97` "Abort Task", `actions/tasks.js:117`, drawer `TaskDetailActions.tsx:112` — 15/15 abort checks PASS; task Rollback route/`@CanRollback` absent (C16/C17 FAIL). | fixed (abort delivered; rollback half not on this branch) |
| 2025.1 | 2025.1.4.0-b103 (tag v2025.1.4.0) | 690ac4141e57d060fd834312f396643e8c41f260 | Runtime repro not applicable (`NO_YBA`, digest OK, revision = tag SHA). Source at tag: route `v1.routes:468`, controller `:572`, commissioner `:250`, executor `:483`, `ITask.java:35`, `TaskInfo.java:108`, `TaskListTable.js:87`, `tasks.js:105`, drawer `:113`, rollback route `:470` + `@CanRollback` `:28` — 17/17 PASS. | fixed |
| 2025.2 | 2025.2.6.0-b111 (tag v2025.2.6.0) | ae22ef35ff65de63303467d95ac3e6d2382feba9 | Runtime repro not applicable (`NO_YBA`, digest OK, revision = tag SHA). Source at tag (raw.githubusercontent.com at this SHA): route `v1.routes:471`, controller `:615`, commissioner `:275`, executor `:494`, `ITask.java:36`, `TaskInfo.java:109`, `TaskListTable.js:87`, `tasks.js:105`, drawer `:113`, rollback route `:473` + `@CanRollback` `:29` — 17/17 PASS. | fixed |
| 2026.1 | 2026.1.1.1-b2 (tag v2026.1.1.1; = `latest`) | 3b010e021f6027744bf9e016b17e108a2eb24e44 | Runtime repro not applicable (`NO_YBA`, digest OK, revision = tag SHA). Source at tag: route `v1.routes:472`, controller `:629`, commissioner `:256`, executor `:495`, `ITask.java:36`, `TaskInfo.java:115`, `TaskListTable.jsx:87`, `tasks.js:105`, drawer `:151`, rollback route `:474` + `@CanRollback` `:29` — 17/17 PASS; 67 `@Abortable` task classes. | fixed |
| master | preview 2.25.2.0-b359 (tag v2.25.2.0); master HEAD b8477c42 | 03bbc9f31eae5c3fecabd8f0afa5e99211e6080d (HEAD b8477c4284908e7ff9314fb2b31014d12f20cc0c) | Runtime repro not applicable (`NO_YBA`, digest OK, revision = tag SHA). Source at tag: route `v1.routes:468`, controller `:572`, commissioner `:235`, executor `:459`, `ITask.java:35`, `TaskInfo.java:101`, `TaskListTable.js:87`, `tasks.js:105`, drawer `:113`, rollback route `:470` + `@CanRollback` `:28` — 17/17 PASS. Master checkout 46a9f7adfd carries the same mechanism (citations in Root cause). | fixed |

Reported release: none stated (2020-05 platform; pre-dates the feature). Not separately testable.

First-fixed provenance (tag/commit ancestry; separate from the runtime/source evidence above):
implementing commits on master — `79db6df6d378bd43ce444b4d483b3c14b8e91f1d` 2021-12-15
"[PLAT-2152] ... ability to send abort signal to a task" (D13797);
`97757d26a98c0bf80e72bbebd905e10af754600c` 2021-12-18 "[PLAT-2535] Add API endpoint for aborting
tasks" (D14391); `77f3bc949353db74238357e8b35df0388d251104` 2022-01-01 "[PLAT-2562] UI support for
abort task capability" (D14528); `8bd0b1812d530c8b39411c69f9311b4ba66d7597` 2022-01-06
"[PLAT-2572] ... abortable in CustomerTask API response payload" (+ `@Abortable`/`@Retryable`
annotations, D14537). All four are reachable from tag `v2.11.2`
(d142556567b5e1c83ea5c915ec7b9964492b2321) and tag `v2.12.1`
(eb518bf59e54d12a1cca20f393b78c3186e7c2f8) per `github:list_commits sha=<tag> path=<file>`.
Official release notes list PLAT-2535/PLAT-2562 under **v2.11.2.0 (2022-01-26, build
2.11.2.0-b89)** — first preview release — and **v2.12.1.0 (2022-02-22, build 2.12.1.0-b41)** —
first stable release (`docs/content/stable/releases/ybdb-releases/end-of-life/v2.11.md:62-64,84,96`,
`.../v2.12.md:1238-1240`). Every in-support branch (2024.2, 2025.1, 2025.2, 2026.1) post-dates
2.12, so no backport is required. The separate task Rollback API (revert half) first appears at
`v2.25.1.0` (preview) / `v2025.1.0.0` (stable); absent at `v2024.2.0.0`, `v2024.2.11.0`, `v2.25.0.0`.

## Proposed fix direction
No code change needed for the issue's core ask. Recommended disposition: close #4568 as delivered
(PLAT-2535 API, PLAT-2562 UI, first shipped v2.11.2.0 / v2.12.1.0), noting that the "revert YW
metadata" clause is served by the separate Retry and Rollback task actions (Rollback on 2025.1+
for `@CanRollback` task types). If a maintainer wants the literal "Actions dropdown", that is a
cosmetic UI restructuring of `TaskListTable.jsx:64-103` (the legacy list already exposes the
action inline; the redesigned drawer groups Retry/Rollback/Abort as buttons). No backport to
2024.2 of the Rollback API is warranted by this issue.

## Test strategy
Existing coverage: `managed/src/test/java/com/yugabyte/yw/commissioner/TaskExecutorTest.java`
(`testSubTaskNonAbortable` `:368`, `testSubTaskAbort` `:390`, `testSubTaskAbortAtPosition` `:453`)
and `managed/src/test/java/com/yugabyte/yw/controllers/CustomerTaskControllerTest.java`
(`testRetryAndAbortResolveUniverseFromTaskParams` `:880`, abort-endpoint RBAC resolution). If the
UI is restructured, extend the
Tasks-page component tests under `managed/ui/src/components/tasks/`. Public entry points to
exercise manually: `POST /api/v1/customers/{cUUID}/tasks/{tUUID}/abort` and the task-list payload
fields `abortable`, `retryable`, `canRollback`, `status` (`Abort` → `Aborted`).

## Confidentiality check
Contains only public GitHub issue text, public commit SHAs/subjects, public release-note text,
public Docker Hub metadata, and source paths from the open-source repository. No customer data,
PII, internal hostnames, credentials, or non-public tracker content (PLAT-/Aha identifiers appear
verbatim in public commits and release notes).


[DEVOPS-3852]: https://yugabyte.atlassian.net/browse/DEVOPS-3852?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by kriteshranjan** [comment]
AI validation notes (`yb4568-tag-verification.txt`):

# yb#4568 per-tag mechanism verification — 46a9f7adfd checkout; run at 2026-09-05T19:21:18Z
# Format: <target> <tag> <sha> <check> <PASS|FAIL|NOFILE> <first-match path:line>
== 2024.2 v2024.2.11.0 04544fd94fb4e50d05994784f444b075a4a5dfdb (local-git) ==
2024.2 v2024.2.11.0 04544fd94fb4e50d05994784f444b075a4a5dfdb C1 PASS managed/src/main/resources/v1.routes:460  # REST route POST /customers/:cUUID/tasks/:tUUID/abort
2024.2 v2024.2.11.0 04544fd94fb4e50d05994784f444b075a4a5dfdb C2 PASS managed/src/main/java/com/yugabyte/yw/controllers/CustomerTaskController.java:394  # controller abortTask handler
2024.2 v2024.2.11.0 04544fd94fb4e50d05994784f444b075a4a5dfdb C3 PASS managed/src/main/java/com/yugabyte/yw/controllers/CustomerTaskController.java:404  # audit entry ActionType.Abort on abort
2024.2 v2024.2.11.0 04544fd94fb4e50d05994784f444b075a4a5dfdb C4 PASS managed/src/main/java/com/yugabyte/yw/commissioner/Commissioner.java:170  # Commissioner.abortTask
2024.2 v2024.2.11.0 04544fd94fb4e50d05994784f444b075a4a5dfdb C5 PASS managed/src/main/java/com/yugabyte/yw/commissioner/Commissioner.java:289  # task list payload exposes abortable flag (arg shape differs by branch)
2024.2 v2024.2.11.0 04544fd94fb4e50d05994784f444b075a4a5dfdb C6 PASS managed/src/main/java/com/yugabyte/yw/commissioner/TaskExecutor.java:452  # TaskExecutor.abort signals RunnableTask
2024.2 v2024.2.11.0 04544fd94fb4e50d05994784f444b075a4a5dfdb C7 PASS managed/src/main/java/com/yugabyte/yw/commissioner/TaskExecutor.java:749  # executor persists terminal Aborted state on CancellationException
2024.2 v2024.2.11.0 04544fd94fb4e50d05994784f444b075a4a5dfdb C8 PASS managed/src/main/java/com/yugabyte/yw/commissioner/ITask.java:26  # @Abortable task-class annotation
2024.2 v2024.2.11.0 04544fd94fb4e50d05994784f444b075a4a5dfdb C9 PASS managed/src/main/java/com/yugabyte/yw/models/TaskInfo.java:102  # TaskInfo.State.Aborted enum value
2024.2 v2024.2.11.0 04544fd94fb4e50d05994784f444b075a4a5dfdb C10 PASS managed/src/main/java/com/yugabyte/yw/models/TaskInfo.java:99  # TaskInfo.State.Abort (in-progress) enum value
2024.2 v2024.2.11.0 04544fd94fb4e50d05994784f444b075a4a5dfdb C11 PASS managed/src/main/java/com/yugabyte/yw/models/CustomerTask.java:507  # CustomerTask status text 'Task aborted'/'Aborting task'
2024.2 v2024.2.11.0 04544fd94fb4e50d05994784f444b075a4a5dfdb C12 PASS managed/ui/src/components/tasks/TaskList/TaskListTable.js:97  # Tasks page per-row 'Abort Task' action (legacy UI; .jsx on 2026.1, .js earlier)
2024.2 v2024.2.11.0 04544fd94fb4e50d05994784f444b075a4a5dfdb C13 PASS managed/ui/src/components/tasks/TaskList/TaskListTable.js:77  # Tasks page gates action on abortable flag
2024.2 v2024.2.11.0 04544fd94fb4e50d05994784f444b075a4a5dfdb C14 PASS managed/ui/src/actions/tasks.js:117  # UI action posts to /tasks/<uuid>/abort
2024.2 v2024.2.11.0 04544fd94fb4e50d05994784f444b075a4a5dfdb C15 PASS managed/ui/src/redesign/features/tasks/components/drawerComp/TaskDetailActions.tsx:112  # redesigned task drawer Abort button
2024.2 v2024.2.11.0 04544fd94fb4e50d05994784f444b075a4a5dfdb C16 FAIL managed/src/main/resources/v1.routes  # REST route POST /customers/:cUUID/tasks/:tUUID/rollback (revert-metadata half)
2024.2 v2024.2.11.0 04544fd94fb4e50d05994784f444b075a4a5dfdb C17 FAIL managed/src/main/java/com/yugabyte/yw/commissioner/ITask.java  # @CanRollback task-class annotation
2024.2 v2024.2.11.0 04544fd94fb4e50d05994784f444b075a4a5dfdb SUMMARY pass=15 fail=2 abortable_task_classes=39
== 2025.1 v2025.1.4.0 690ac4141e57d060fd834312f396643e8c41f260 (local-git) ==
2025.1 v2025.1.4.0 690ac4141e57d060fd834312f396643e8c41f260 C1 PASS managed/src/main/resources/v1.routes:468  # REST route POST /customers/:cUUID/tasks/:tUUID/abort
2025.1 v2025.1.4.0 690ac4141e57d060fd834312f396643e8c41f260 C2 PASS managed/src/main/java/com/yugabyte/yw/controllers/CustomerTaskController.java:572  # controller abortTask handler
2025.1 v2025.1.4.0 690ac4141e57d060fd834312f396643e8c41f260 C3 PASS managed/src/main/java/com/yugabyte/yw/controllers/CustomerTaskController.java:582  # audit entry ActionType.Abort on abort
2025.1 v2025.1.4.0 690ac4141e57d060fd834312f396643e8c41f260 C4 PASS managed/src/main/java/com/yugabyte/yw/commissioner/Commissioner.java:250  # Commissioner.abortTask
2025.1 v2025.1.4.0 690ac4141e57d060fd834312f396643e8c41f260 C5 PASS managed/src/main/java/com/yugabyte/yw/commissioner/Commissioner.java:368  # task list payload exposes abortable flag (arg shape differs by branch)
2025.1 v2025.1.4.0 690ac4141e57d060fd834312f396643e8c41f260 C6 PASS managed/src/main/java/com/yugabyte/yw/commissioner/TaskExecutor.java:483  # TaskExecutor.abort signals RunnableTask
2025.1 v2025.1.4.0 690ac4141e57d060fd834312f396643e8c41f260 C7 PASS managed/src/main/java/com/yugabyte/yw/commissioner/TaskExecutor.java:774  # executor persists terminal Aborted state on CancellationException
2025.1 v2025.1.4.0 690ac4141e57d060fd834312f396643e8c41f260 C8 PASS managed/src/main/java/com/yugabyte/yw/commissioner/ITask.java:35  # @Abortable task-class annotation
2025.1 v2025.1.4.0 690ac4141e57d060fd834312f396643e8c41f260 C9 PASS managed/src/main/java/com/yugabyte/yw/models/TaskInfo.java:108  # TaskInfo.State.Aborted enum value
2025.1 v2025.1.4.0 690ac4141e57d060fd834312f396643e8c41f260 C10 PASS managed/src/main/java/com/yugabyte/yw/models/TaskInfo.java:105  # TaskInfo.State.Abort (in-progress) enum value
2025.1 v2025.1.4.0 690ac4141e57d060fd834312f396643e8c41f260 C11 PASS managed/src/main/java/com/yugabyte/yw/models/CustomerTask.java:532  # CustomerTask status text 'Task aborted'/'Aborting task'
2025.1 v2025.1.4.0 690ac4141e57d060fd834312f396643e8c41f260 C12 PASS managed/ui/src/components/tasks/TaskList/TaskListTable.js:87  # Tasks page per-row 'Abort Task' action (legacy UI; .jsx on 2026.1, .js earlier)
2025.1 v2025.1.4.0 690ac4141e57d060fd834312f396643e8c41f260 C13 PASS managed/ui/src/components/tasks/TaskList/TaskListTable.js:76  # Tasks page gates action on abortable flag
2025.1 v2025.1.4.0 690ac4141e57d060fd834312f396643e8c41f260 C14 PASS managed/ui/src/actions/tasks.js:105  # UI action posts to /tasks/<uuid>/abort
2025.1 v2025.1.4.0 690ac4141e57d060fd834312f396643e8c41f260 C15 PASS managed/ui/src/redesign/features/tasks/components/drawerComp/TaskDetailActions.tsx:113  # redesigned task drawer Abort button
2025.1 v2025.1.4.0 690ac4141e57d060fd834312f396643e8c41f260 C16 PASS managed/src/main/resources/v1.routes:470  # REST route POST /customers/:cUUID/tasks/:tUUID/rollback (revert-metadata half)
2025.1 v2025.1.4.0 690ac4141e57d060fd834312f396643e8c41f260 C17 PASS managed/src/main/java/com/yugabyte/yw/commissioner/ITask.java:28  # @CanRollback task-class annotation
2025.1 v2025.1.4.0 690ac4141e57d060fd834312f396643e8c41f260 SUMMARY pass=17 fail=0 abortable_task_classes=52
== 2025.2 v2025.2.6.0 ae22ef35ff65de63303467d95ac3e6d2382feba9 (raw-github) ==
2025.2 v2025.2.6.0 ae22ef35ff65de63303467d95ac3e6d2382feba9 C1 PASS managed/src/main/resources/v1.routes:471  # REST route POST /customers/:cUUID/tasks/:tUUID/abort
2025.2 v2025.2.6.0 ae22ef35ff65de63303467d95ac3e6d2382feba9 C2 PASS managed/src/main/java/com/yugabyte/yw/controllers/CustomerTaskController.java:615  # controller abortTask handler
2025.2 v2025.2.6.0 ae22ef35ff65de63303467d95ac3e6d2382feba9 C3 PASS managed/src/main/java/com/yugabyte/yw/controllers/CustomerTaskController.java:625  # audit entry ActionType.Abort on abort
2025.2 v2025.2.6.0 ae22ef35ff65de63303467d95ac3e6d2382feba9 C4 PASS managed/src/main/java/com/yugabyte/yw/commissioner/Commissioner.java:275  # Commissioner.abortTask
2025.2 v2025.2.6.0 ae22ef35ff65de63303467d95ac3e6d2382feba9 C5 PASS managed/src/main/java/com/yugabyte/yw/commissioner/Commissioner.java:394  # task list payload exposes abortable flag (arg shape differs by branch)
2025.2 v2025.2.6.0 ae22ef35ff65de63303467d95ac3e6d2382feba9 C6 PASS managed/src/main/java/com/yugabyte/yw/commissioner/TaskExecutor.java:494  # TaskExecutor.abort signals RunnableTask
2025.2 v2025.2.6.0 ae22ef35ff65de63303467d95ac3e6d2382feba9 C7 PASS managed/src/main/java/com/yugabyte/yw/commissioner/TaskExecutor.java:794  # executor persists terminal Aborted state on CancellationException
2025.2 v2025.2.6.0 ae22ef35ff65de63303467d95ac3e6d2382feba9 C8 PASS managed/src/main/java/com/yugabyte/yw/commissioner/ITask.java:36  # @Abortable task-class annotation
2025.2 v2025.2.6.0 ae22ef35ff65de63303467d95ac3e6d2382feba9 C9 PASS managed/src/main/java/com/yugabyte/yw/models/TaskInfo.java:109  # TaskInfo.State.Aborted enum value
2025.2 v2025.2.6.0 ae22ef35ff65de63303467d95ac3e6d2382feba9 C10 PASS managed/src/main/java/com/yugabyte/yw/models/TaskInfo.java:106  # TaskInfo.State.Abort (in-progress) enum value
2025.2 v2025.2.6.0 ae22ef35ff65de63303467d95ac3e6d2382feba9 C11 PASS managed/src/main/java/com/yugabyte/yw/models/CustomerTask.java:565  # CustomerTask status text 'Task aborted'/'Aborting task'
2025.2 v2025.2.6.0 ae22ef35ff65de63303467d95ac3e6d2382feba9 C12 PASS managed/ui/src/components/tasks/TaskList/TaskListTable.js:87  # Tasks page per-row 'Abort Task' action (legacy UI; .jsx on 2026.1, .js earlier)
2025.2 v2025.2.6.0 ae22ef35ff65de63303467d95ac3e6d2382feba9 C13 PASS managed/ui/src/components/tasks/TaskList/TaskListTable.js:76  # Tasks page gates action on abortable flag
2025.2 v2025.2.6.0 ae22ef35ff65de63303467d95ac3e6d2382feba9 C14 PASS managed/ui/src/actions/tasks.js:105  # UI action posts to /tasks/<uuid>/abort
2025.2 v2025.2.6.0 ae22ef35ff65de63303467d95ac3e6d2382feba9 C15 PASS managed/ui/src/redesign/features/tasks/components/drawerComp/TaskDetailActions.tsx:113  # redesigned task drawer Abort button
2025.2 v2025.2.6.0 ae22ef35ff65de63303467d95ac3e6d2382feba9 C16 PASS managed/src/main/resources/v1.routes:473  # REST route POST /customers/:cUUID/tasks/:tUUID/rollback (revert-metadata half)
2025.2 v2025.2.6.0 ae22ef35ff65de63303467d95ac3e6d2382feba9 C17 PASS managed/src/main/java/com/yugabyte/yw/commissioner/ITask.java:29  # @CanRollback task-class annotation
2025.2 v2025.2.6.0 ae22ef35ff65de63303467d95ac3e6d2382feba9 SUMMARY pass=17 fail=0 abortable_task_classes=n/a(raw-mode)
== 2026.1 v2026.1.1.1 3b010e021f6027744bf9e016b17e108a2eb24e44 (local-git) ==
2026.1 v2026.1.1.1 3b010e021f6027744bf9e016b17e108a2eb24e44 C1 PASS managed/src/main/resources/v1.routes:472  # REST route POST /customers/:cUUID/tasks/:tUUID/abort
2026.1 v2026.1.1.1 3b010e021f6027744bf9e016b17e108a2eb24e44 C2 PASS managed/src/main/java/com/yugabyte/yw/controllers/CustomerTaskController.java:629  # controller abortTask handler
2026.1 v2026.1.1.1 3b010e021f6027744bf9e016b17e108a2eb24e44 C3 PASS managed/src/main/java/com/yugabyte/yw/controllers/CustomerTaskController.java:639  # audit entry ActionType.Abort on abort
2026.1 v2026.1.1.1 3b010e021f6027744bf9e016b17e108a2eb24e44 C4 PASS managed/src/main/java/com/yugabyte/yw/commissioner/Commissioner.java:256  # Commissioner.abortTask
2026.1 v2026.1.1.1 3b010e021f6027744bf9e016b17e108a2eb24e44 C5 PASS managed/src/main/java/com/yugabyte/yw/commissioner/Commissioner.java:383  # task list payload exposes abortable flag (arg shape differs by branch)
2026.1 v2026.1.1.1 3b010e021f6027744bf9e016b17e108a2eb24e44 C6 PASS managed/src/main/java/com/yugabyte/yw/commissioner/TaskExecutor.java:495  # TaskExecutor.abort signals RunnableTask
2026.1 v2026.1.1.1 3b010e021f6027744bf9e016b17e108a2eb24e44 C7 PASS managed/src/main/java/com/yugabyte/yw/commissioner/TaskExecutor.java:797  # executor persists terminal Aborted state on CancellationException
2026.1 v2026.1.1.1 3b010e021f6027744bf9e016b17e108a2eb24e44 C8 PASS managed/src/main/java/com/yugabyte/yw/commissioner/ITask.java:36  # @Abortable task-class annotation
2026.1 v2026.1.1.1 3b010e021f6027744bf9e016b17e108a2eb24e44 C9 PASS managed/src/main/java/com/yugabyte/yw/models/TaskInfo.java:115  # TaskInfo.State.Aborted enum value
2026.1 v2026.1.1.1 3b010e021f6027744bf9e016b17e108a2eb24e44 C10 PASS managed/src/main/java/com/yugabyte/yw/models/TaskInfo.java:112  # TaskInfo.State.Abort (in-progress) enum value
2026.1 v2026.1.1.1 3b010e021f6027744bf9e016b17e108a2eb24e44 C11 PASS managed/src/main/java/com/yugabyte/yw/models/CustomerTask.java:578  # CustomerTask status text 'Task aborted'/'Aborting task'
2026.1 v2026.1.1.1 3b010e021f6027744bf9e016b17e108a2eb24e44 C12 PASS managed/ui/src/components/tasks/TaskList/TaskListTable.jsx:87  # Tasks page per-row 'Abort Task' action (legacy UI; .jsx on 2026.1, .js earlier)
2026.1 v2026.1.1.1 3b010e021f6027744bf9e016b17e108a2eb24e44 C13 PASS managed/ui/src/components/tasks/TaskList/TaskListTable.jsx:76  # Tasks page gates action on abortable flag
2026.1 v2026.1.1.1 3b010e021f6027744bf9e016b17e108a2eb24e44 C14 PASS managed/ui/src/actions/tasks.js:105  # UI action posts to /tasks/<uuid>/abort
2026.1 v2026.1.1.1 3b010e021f6027744bf9e016b17e108a2eb24e44 C15 PASS managed/ui/src/redesign/features/tasks/components/drawerComp/TaskDetailActions.tsx:151  # redesigned task drawer Abort button
2026.1 v2026.1.1.1 3b010e021f6027744bf9e016b17e108a2eb24e44 C16 PASS managed/src/main/resources/v1.routes:474  # REST route POST /customers/:cUUID/tasks/:tUUID/rollback (revert-metadata half)
2026.1 v2026.1.1.1 3b010e021f6027744bf9e016b17e108a2eb24e44 C17 PASS managed/src/main/java/com/yugabyte/yw/commissioner/ITask.java:29  # @CanRollback task-class annotation
2026.1 v2026.1.1.1 3b010e021f6027744bf9e016b17e108a2eb24e44 SUMMARY pass=17 fail=0 abortable_task_classes=67
== master v2.25.2.0 03bbc9f31eae5c3fecabd8f0afa5e99211e6080d (local-git) ==
master v2.25.2.0 03bbc9f31eae5c3fecabd8f0afa5e99211e6080d C1 PASS managed/src/main/resources/v1.routes:468  # REST route POST /customers/:cUUID/tasks/:tUUID/abort
master v2.25.2.0 03bbc9f31eae5c3fecabd8f0afa5e99211e6080d C2 PASS managed/src/main/java/com/yugabyte/yw/controllers/CustomerTaskController.java:572  # controller abortTask handler
master v2.25.2.0 03bbc9f31eae5c3fecabd8f0afa5e99211e6080d C3 PASS managed/src/main/java/com/yugabyte/yw/controllers/CustomerTaskController.java:582  # audit entry ActionType.Abort on abort
master v2.25.2.0 03bbc9f31eae5c3fecabd8f0afa5e99211e6080d C4 PASS managed/src/main/java/com/yugabyte/yw/commissioner/Commissioner.java:235  # Commissioner.abortTask
master v2.25.2.0 03bbc9f31eae5c3fecabd8f0afa5e99211e6080d C5 PASS managed/src/main/java/com/yugabyte/yw/commissioner/Commissioner.java:353  # task list payload exposes abortable flag (arg shape differs by branch)
master v2.25.2.0 03bbc9f31eae5c3fecabd8f0afa5e99211e6080d C6 PASS managed/src/main/java/com/yugabyte/yw/commissioner/TaskExecutor.java:459  # TaskExecutor.abort signals RunnableTask
master v2.25.2.0 03bbc9f31eae5c3fecabd8f0afa5e99211e6080d C7 PASS managed/src/main/java/com/yugabyte/yw/commissioner/TaskExecutor.java:731  # executor persists terminal Aborted state on CancellationException
master v2.25.2.0 03bbc9f31eae5c3fecabd8f0afa5e99211e6080d C8 PASS managed/src/main/java/com/yugabyte/yw/commissioner/ITask.java:35  # @Abortable task-class annotation
master v2.25.2.0 03bbc9f31eae5c3fecabd8f0afa5e99211e6080d C9 PASS managed/src/main/java/com/yugabyte/yw/models/TaskInfo.java:101  # TaskInfo.State.Aborted enum value
master v2.25.2.0 03bbc9f31eae5c3fecabd8f0afa5e99211e6080d C10 PASS managed/src/main/java/com/yugabyte/yw/models/TaskInfo.java:98  # TaskInfo.State.Abort (in-progress) enum value
master v2.25.2.0 03bbc9f31eae5c3fecabd8f0afa5e99211e6080d C11 PASS managed/src/main/java/com/yugabyte/yw/models/CustomerTask.java:532  # CustomerTask status text 'Task aborted'/'Aborting task'
master v2.25.2.0 03bbc9f31eae5c3fecabd8f0afa5e99211e6080d C12 PASS managed/ui/src/components/tasks/TaskList/TaskListTable.js:87  # Tasks page per-row 'Abort Task' action (legacy UI; .jsx on 2026.1, .js earlier)
master v2.25.2.0 03bbc9f31eae5c3fecabd8f0afa5e99211e6080d C13 PASS managed/ui/src/components/tasks/TaskList/TaskListTable.js:76  # Tasks page gates action on abortable flag
master v2.25.2.0 03bbc9f31eae5c3fecabd8f0afa5e99211e6080d C14 PASS managed/ui/src/actions/tasks.js:105  # UI action posts to /tasks/<uuid>/abort
master v2.25.2.0 03bbc9f31eae5c3fecabd8f0afa5e99211e6080d C15 PASS managed/ui/src/redesign/features/tasks/components/drawerComp/TaskDetailActions.tsx:113  # redesigned task drawer Abort button
master v2.25.2.0 03bbc9f31eae5c3fecabd8f0afa5e99211e6080d C16 PASS managed/src/main/resources/v1.routes:470  # REST route POST /customers/:cUUID/tasks/:tUUID/rollback (revert-metadata half)
master v2.25.2.0 03bbc9f31eae5c3fecabd8f0afa5e99211e6080d C17 PASS managed/src/main/java/com/yugabyte/yw/commissioner/ITask.java:28  # @CanRollback task-class annotation
master v2.25.2.0 03bbc9f31eae5c3fecabd8f0afa5e99211e6080d SUMMARY pass=17 fail=0 abortable_task_classes=52

**Comment by kriteshranjan** [comment]
AI validation notes (`yb4568-image-check.txt`):

# run at 2026-09-05T19:24:15Z via podman 5.8.6
== 2024.2 docker.io/yugabytedb/yugabyte:2024.2.11.0-b36-aarch64 ==
2024.2 DIGEST OK sha256:53c01c56315e245cc901cbc46e93f30fedb1f87833c7b1aa09d20fbeda27a288
2024.2 yb-master --version: version 2024.2.11.0 build 36 revision 04544fd94fb4e50d05994784f444b075a4a5dfdb build_type RELEASE built at 18 Aug 2026 00:58:07 UTC
2024.2 DB_BINARY_PRESENT /home/yugabyte/bin/yb-master
2024.2 DB_BINARY_PRESENT /home/yugabyte/bin/yb-tserver
2024.2 DB_BINARY_PRESENT /home/yugabyte/bin/yugabyted
2024.2 DB_BINARY_PRESENT /home/yugabyte/bin/ysqlsh
2024.2 NO_YBA: no yugaware/YBA files in image
2024.2 auto_flags.json bin cores gflag_allowlist.txt lib master master_flags.xml openssl-config postgres pylib share tools tserver tserver_flags.xml version_metadata.json www 
== 2025.1 docker.io/yugabytedb/yugabyte:2025.1.4.0-b103-aarch64 ==
2025.1 DIGEST OK sha256:08f1be0d6e074e0286e70067c12933a8cf536876d50fc43c52b1127d447ecdc0
2025.1 yb-master --version: version 2025.1.4.0 build 103 revision 690ac4141e57d060fd834312f396643e8c41f260 build_type RELEASE built at 02 Apr 2026 20:43:14 UTC
2025.1 DB_BINARY_PRESENT /home/yugabyte/bin/yb-master
2025.1 DB_BINARY_PRESENT /home/yugabyte/bin/yb-tserver
2025.1 DB_BINARY_PRESENT /home/yugabyte/bin/yugabyted
2025.1 DB_BINARY_PRESENT /home/yugabyte/bin/ysqlsh
2025.1 NO_YBA: no yugaware/YBA files in image
2025.1 auto_flags.json bin cores gflag_allowlist.txt lib master master_flags.xml openssl-config postgres pylib share tools tserver tserver_flags.xml version_metadata.json www 
== 2025.2 docker.io/yugabytedb/yugabyte:2025.2.6.0-b111-aarch64 ==
2025.2 DIGEST OK sha256:4a1f7a94d3b70c61ecaf5e049521d691b16c9bf204ed01bdb6ab75d5beeb623e
2025.2 yb-master --version: version 2025.2.6.0 build 111 revision ae22ef35ff65de63303467d95ac3e6d2382feba9 build_type RELEASE built at 02 Sep 2026 16:42:37 UTC
2025.2 DB_BINARY_PRESENT /home/yugabyte/bin/yb-master
2025.2 DB_BINARY_PRESENT /home/yugabyte/bin/yb-tserver
2025.2 DB_BINARY_PRESENT /home/yugabyte/bin/yugabyted
2025.2 DB_BINARY_PRESENT /home/yugabyte/bin/ysqlsh
2025.2 NO_YBA: no yugaware/YBA files in image
2025.2 auto_flags.json bin controller cores gflag_allowlist.txt lib master master_flags.xml openssl-config postgres pylib python share tools tserver tserver_flags.xml version_metadata.json www ybc-2.2.0.3-b22-el8-aarch64 
== 2026.1 docker.io/yugabytedb/yugabyte:2026.1.1.1-b2-aarch64 ==
2026.1 DIGEST OK sha256:d27eede72bbc669e348e9920318589b4bb68e98204a241ec6bfaba04bf4f6699
2026.1 yb-master --version: version 2026.1.1.1 build 2 revision 3b010e021f6027744bf9e016b17e108a2eb24e44 build_type RELEASE built at 14 Aug 2026 04:03:14 UTC
2026.1 DB_BINARY_PRESENT /home/yugabyte/bin/yb-master
2026.1 DB_BINARY_PRESENT /home/yugabyte/bin/yb-tserver
2026.1 DB_BINARY_PRESENT /home/yugabyte/bin/yugabyted
2026.1 DB_BINARY_PRESENT /home/yugabyte/bin/ysqlsh
2026.1 NO_YBA: no yugaware/YBA files in image
2026.1 auto_flags.json bin controller cores gflag_allowlist.txt lib master master_flags.xml openssl-config postgres pylib python share tools tserver tserver_flags.xml version_metadata.json www ybc-2.2.0.4-b5-el8-aarch64 
== master docker.io/yugabytedb/yugabyte:2.25.2.0-b359-aarch64 ==
master DIGEST OK sha256:3f9cd1b8f520cc78e82ace51174be73343a551148bc6c9a9f821a22f1d2858c5
master yb-master --version: version 2.25.2.0 build 359 revision 03bbc9f31eae5c3fecabd8f0afa5e99211e6080d build_type RELEASE built at 08 May 2025 19:05:35 UTC
master DB_BINARY_PRESENT /home/yugabyte/bin/yb-master
master DB_BINARY_PRESENT /home/yugabyte/bin/yb-tserver
master DB_BINARY_PRESENT /home/yugabyte/bin/yugabyted
master DB_BINARY_PRESENT /home/yugabyte/bin/ysqlsh
master NO_YBA: no yugaware/YBA files in image
master auto_flags.json bin cores gflag_allowlist.txt lib master master_flags.xml openssl-config postgres pylib share tools tserver tserver_flags.xml version_metadata.json www

**Comment by kriteshranjan** [comment]
AI validation notes (`yb4568-verify-tags-sh.txt`):

#!/usr/bin/env bash
# yugabyte/yugabyte-db#4568 "[Platform] Add ability for an Admin to abort a Task"
# Per-tag SOURCE verification of the YBA task-abort mechanism at the five resolved
# support-matrix targets. YBA (managed/) does not ship in yugabytedb/yugabyte DB
# images, so the matrix is grounded in source inspection at the exact tag SHAs.
#
# Method: `git grep` against locally available tag objects; for a tag missing from
# the shallow checkout the same files are fetched from raw.githubusercontent.com at
# the exact commit SHA and grepped identically. Output: one PASS/FAIL line per check.
#
# Usage: bash yb4568-verify-tags-sh.txt <repo-dir> > yb4568-tag-verification.txt
set -u
REPO="${1:?repo dir}"
WORK="${TMPDIR:-/tmp}/yb4568-verify"
mkdir -p "$WORK"

# target|tag|sha
TARGETS=(
  "2024.2|v2024.2.11.0|04544fd94fb4e50d05994784f444b075a4a5dfdb"
  "2025.1|v2025.1.4.0|690ac4141e57d060fd834312f396643e8c41f260"
  "2025.2|v2025.2.6.0|ae22ef35ff65de63303467d95ac3e6d2382feba9"
  "2026.1|v2026.1.1.1|3b010e021f6027744bf9e016b17e108a2eb24e44"
  "master|v2.25.2.0|03bbc9f31eae5c3fecabd8f0afa5e99211e6080d"
)

# check-id|file|ERE pattern|meaning
CHECKS=(
  "C1|managed/src/main/resources/v1.routes|^POST[[:space:]]+/customers/:cUUID/tasks/:tUUID/abort[[:space:]]+com\.yugabyte\.yw\.controllers\.CustomerTaskController\.abortTask|REST route POST /customers/:cUUID/tasks/:tUUID/abort"
  "C2|managed/src/main/java/com/yugabyte/yw/controllers/CustomerTaskController.java|public Result abortTask\(UUID customerUUID, UUID taskUUID|controller abortTask handler"
  "C3|managed/src/main/java/com/yugabyte/yw/controllers/CustomerTaskController.java|Audit\.ActionType\.Abort|audit entry ActionType.Abort on abort"
  "C4|managed/src/main/java/com/yugabyte/yw/commissioner/Commissioner.java|public boolean abortTask\(UUID taskUUID, boolean force\)|Commissioner.abortTask"
  "C5|managed/src/main/java/com/yugabyte/yw/commissioner/Commissioner.java|responseJson\.put\(\"abortable\", isTaskAbortable\(|task list payload exposes abortable flag (arg shape differs by branch)"
  "C6|managed/src/main/java/com/yugabyte/yw/commissioner/TaskExecutor.java|public Optional<TaskInfo> abort\(UUID taskUUID, boolean force\)|TaskExecutor.abort signals RunnableTask"
  "C7|managed/src/main/java/com/yugabyte/yw/commissioner/TaskExecutor.java|updateTaskDetailsOnError\(TaskInfo\.State\.Aborted|executor persists terminal Aborted state on CancellationException"
  "C8|managed/src/main/java/com/yugabyte/yw/commissioner/ITask.java|@interface Abortable|@Abortable task-class annotation"
  "C9|managed/src/main/java/com/yugabyte/yw/models/TaskInfo.java|Aborted\([0-9]+\)|TaskInfo.State.Aborted enum value"
  "C10|managed/src/main/java/com/yugabyte/yw/models/TaskInfo.java|^[[:space:]]*Abort\([0-9]+\)|TaskInfo.State.Abort (in-progress) enum value"
  "C11|managed/src/main/java/com/yugabyte/yw/models/CustomerTask.java|Task aborted|CustomerTask status text 'Task aborted'/'Aborting task'"
  "C12|managed/ui/src/components/tasks/TaskList/TaskListTable.jsx,managed/ui/src/components/tasks/TaskList/TaskListTable.js|Abort Task|Tasks page per-row 'Abort Task' action (legacy UI; .jsx on 2026.1, .js earlier)"
  "C13|managed/ui/src/components/tasks/TaskList/TaskListTable.jsx,managed/ui/src/components/tasks/TaskList/TaskListTable.js|row\.abortable|Tasks page gates action on abortable flag"
  "C14|managed/ui/src/actions/tasks.js|/tasks/\\\$\{taskUUID\}/abort|UI action posts to /tasks/<uuid>/abort"
  "C15|managed/ui/src/redesign/features/tasks/components/drawerComp/TaskDetailActions.tsx|data-testid=\"abort-task\"|redesigned task drawer Abort button"
  "C16|managed/src/main/resources/v1.routes|^POST[[:space:]]+/customers/:cUUID/tasks/:tUUID/rollback[[:space:]]+com\.yugabyte\.yw\.controllers\.CustomerTaskController\.rollbackTask|REST route POST /customers/:cUUID/tasks/:tUUID/rollback (revert-metadata half)"
  "C17|managed/src/main/java/com/yugabyte/yw/commissioner/ITask.java|@interface CanRollback|@CanRollback task-class annotation"
)

have_local_tag() { git -C "$REPO" rev-parse -q --verify "$1^{commit}" >/dev/null 2>&1; }

fetch_raw() { # sha path -> local file path (cached)
  local sha="$1" path="$2" out="$WORK/$1/$2"
  if [ ! -s "$out" ]; then
    mkdir -p "$(dirname "$out")"
    curl -sS --max-time 60 -o "$out" -w "%{http_code}" \
      "https://raw.githubusercontent.com/yugabyte/yugabyte-db/$sha/$path" > "$out.code" || true
  fi
  printf '%s' "$out"
}

echo "# yb#4568 per-tag mechanism verification — $(git -C "$REPO" rev-parse --short HEAD 2>/dev/null) checkout; run at $(python3 -c 'import datetime;print(datetime.datetime.now(datetime.timezone.utc).strftime("%Y-%m-%dT%H:%M:%SZ"))')"
echo "# Format: <target> <tag> <sha> <check> <PASS|FAIL|NOFILE> <first-match path:line>"
for entry in "${TARGETS[@]}"; do
  IFS='|' read -r target tag sha <<<"$entry"
  if have_local_tag "$tag"; then
    local_sha=$(git -C "$REPO" rev-parse "$tag^{commit}")
    mode="local-git"
    [ "$local_sha" = "$sha" ] || { echo "$target $tag $sha SHA-MISMATCH local=$local_sha"; continue; }
  else
    mode="raw-github"
  fi
  echo "== $target $tag $sha ($mode) =="
  pass=0; fail=0
  for c in "${CHECKS[@]}"; do
    IFS='|' read -r cid files pat meaning <<<"$c"
    # pick the first candidate filename that exists at this sha (handles .js -> .jsx renames)
    file=""
    IFS=',' read -r -a cands <<<"$files"
    for cand in "${cands[@]}"; do
      if [ "$mode" = "local-git" ]; then
        git -C "$REPO" cat-file -e "$sha:$cand" 2>/dev/null && { file="$cand"; break; }
      else
        f=$(fetch_raw "$sha" "$cand")
        [ "$(cat "$f.code" 2>/dev/null)" = "200" ] && { file="$cand"; break; }
      fi
    done
    if [ -z "$file" ]; then
      echo "$target $tag $sha $cid NOFILE $files  # $meaning"; fail=$((fail+1)); continue
    fi
    if [ "$mode" = "local-git" ]; then
      hit=$(git -C "$REPO" grep -n -E -e "$pat" "$sha" -- "$file" 2>/dev/null | head -1 | sed "s/^$sha://")
    else
      f=$(fetch_raw "$sha" "$file")
      hit=$(grep -n -E -e "$pat" "$f" 2>/dev/null | head -1 | sed "s#^#$file:#")
    fi
    if [ -n "$hit" ]; then
      echo "$target $tag $sha $cid PASS ${hit%%:*}:$(echo "$hit" | cut -d: -f2)  # $meaning"; pass=$((pass+1))
    else
      echo "$target $tag $sha $cid FAIL $file  # $meaning"; fail=$((fail+1))
    fi
  done
  # breadth: how many task classes are annotated @Abortable at this tag
  if [ "$mode" = "local-git" ]; then
    n=$(git -C "$REPO" grep -l -E -e '^@(ITask\.)?Abortable' "$sha" -- 'managed/src/main/java/com/yugabyte/yw/commissioner/tasks/' 2>/dev/null | wc -l | tr -d ' ')
  else
    n="n/a(raw-mode)"
  fi
  echo "$target $tag $sha SUMMARY pass=$pass fail=$fail abortable_task_classes=$n"
done

**Comment by kriteshranjan** [comment]
AI validation notes (`yb4568-dockerhub-tags.txt`):

=== https://hub.docker.com/v2/repositories/yugabytedb/yugabyte/tags?name=2024.2&page_size=4&ordering=last_updated ===
count: 87
  2024.2.11.0-b36  pushed=2026-08-25T16:02:12.738387Z  manifest=sha256:37199e640e1776e59271d585e1af432889172c96ffbc45b4f9c00130af7a0079  amd64=sha256:6bb00eeb3dd97b93f36716080d2cfe96d1e61f864201d9a36ae6e3aeacb6d9ce  arm64=sha256:53c01c56315e245cc901cbc46e93f30fedb1f87833c7b1aa09d20fbeda27a288
  2024.2.11.0-b36-x86_64  pushed=2026-08-25T16:02:08.918001Z  manifest=sha256:6bb00eeb3dd97b93f36716080d2cfe96d1e61f864201d9a36ae6e3aeacb6d9ce  amd64=sha256:6bb00eeb3dd97b93f36716080d2cfe96d1e61f864201d9a36ae6e3aeacb6d9ce  arm64=None
  2024.2.11.0-b36-aarch64  pushed=2026-08-25T15:59:27.917385Z  manifest=sha256:53c01c56315e245cc901cbc46e93f30fedb1f87833c7b1aa09d20fbeda27a288  amd64=None  arm64=sha256:53c01c56315e245cc901cbc46e93f30fedb1f87833c7b1aa09d20fbeda27a288
  2024.2.10.0-b62  pushed=2026-07-09T15:44:45.531365Z  manifest=sha256:d11048c4ad433ad1328fb88fb596ccd6d4a2e55cdb27db940d147b045c26b551  amd64=sha256:5c14c2f89bd796d454be1bb27b100626a177ae34878d37bd52c0b3505083f1cc  arm64=sha256:d962e3d488eb0c4635788fce2e08fe72b9c6becc237ae781b497ae3cf5b77055
=== https://hub.docker.com/v2/repositories/yugabytedb/yugabyte/tags?name=2025.1&page_size=4&ordering=last_updated ===
count: 36
  2025.1.4.0-b103  pushed=2026-04-20T15:59:42.898497Z  manifest=sha256:be935be963ead918d3ba6497505f0b7adb203313ccad710c111f64f697452ede  amd64=sha256:fa5555536e8fe7d9522fb6c9c55f3c6062549f318b79c5902726097549263771  arm64=sha256:08f1be0d6e074e0286e70067c12933a8cf536876d50fc43c52b1127d447ecdc0
  2025.1.4.0-b103-x86_64  pushed=2026-04-20T15:59:39.044404Z  manifest=sha256:fa5555536e8fe7d9522fb6c9c55f3c6062549f318b79c5902726097549263771  amd64=sha256:fa5555536e8fe7d9522fb6c9c55f3c6062549f318b79c5902726097549263771  arm64=None
  2025.1.4.0-b103-aarch64  pushed=2026-04-20T15:57:55.830612Z  manifest=sha256:08f1be0d6e074e0286e70067c12933a8cf536876d50fc43c52b1127d447ecdc0  amd64=None  arm64=sha256:08f1be0d6e074e0286e70067c12933a8cf536876d50fc43c52b1127d447ecdc0
  2025.1.3.2-b1  pushed=2026-02-26T19:01:20.079224Z  manifest=sha256:caff8184ffe0767614c212c70b3e5389f68c78f8bd056c650d22c2f977be3a00  amd64=sha256:965a074a6c424a2ba6c8a195f86c84bb20e909e6b231ab7da6477862efcac4d9  arm64=sha256:8cc0f3be2491f6b884830b95a8d841c451a74fdb8a5db0de04ede48decffa7a1
=== https://hub.docker.com/v2/repositories/yugabytedb/yugabyte/tags?name=2025.2&page_size=4&ordering=last_updated ===
count: 39
  2025.2.6.0-b111  pushed=2026-09-04T22:23:02.776142Z  manifest=sha256:98ad2aa1843f73b79f8f0e5bd9e3c90f0b9d7b63aed7a22eadd72abc2dc23d67  amd64=sha256:5042b5090b49ef49959a4db9d05c90f5da6482aa72265d6ab4a23f8dc9ec0297  arm64=sha256:4a1f7a94d3b70c61ecaf5e049521d691b16c9bf204ed01bdb6ab75d5beeb623e
  2025.2.6.0-b111-x86_64  pushed=2026-09-04T22:22:58.781722Z  manifest=sha256:5042b5090b49ef49959a4db9d05c90f5da6482aa72265d6ab4a23f8dc9ec0297  amd64=sha256:5042b5090b49ef49959a4db9d05c90f5da6482aa72265d6ab4a23f8dc9ec0297  arm64=None
  2025.2.6.0-b111-aarch64  pushed=2026-09-04T22:22:00.73534Z  manifest=sha256:4a1f7a94d3b70c61ecaf5e049521d691b16c9bf204ed01bdb6ab75d5beeb623e  amd64=None  arm64=sha256:4a1f7a94d3b70c61ecaf5e049521d691b16c9bf204ed01bdb6ab75d5beeb623e
  2025.2.5.2-b5  pushed=2026-08-03T18:54:26.310633Z  manifest=sha256:c405a88a0263a7f8526538476de2b8e287e822b381a0a29ce8b2c357c93e4582  amd64=sha256:3dfa310a7a595bbf1ec5e6e6e58e31821aa86a8444b65d800fcc2ef6b3953cbb  arm64=sha256:d23a83c5430fb2f5f0590627f37aa1adeed48d85f64eebedbb2250ced24c5f61
=== https://hub.docker.com/v2/repositories/yugabytedb/yugabyte/tags?name=2026.1&page_size=4&ordering=last_updated ===
count: 12
  2026.1.1.1-b2  pushed=2026-08-18T20:03:16.424245Z  manifest=sha256:3926eedf0ff4b99426fbff533128ff16635ed5697a5cf086dde4fafec28e6f75  amd64=sha256:36741c614b0a28498a3447a01b952240905a0ad87c9f68dbcbb20e65bc3d2e9f  arm64=sha256:d27eede72bbc669e348e9920318589b4bb68e98204a241ec6bfaba04bf4f6699
  2026.1.1.1-b2-x86_64  pushed=2026-08-18T20:03:12.367966Z  manifest=sha256:36741c614b0a28498a3447a01b952240905a0ad87c9f68dbcbb20e65bc3d2e9f  amd64=sha256:36741c614b0a28498a3447a01b952240905a0ad87c9f68dbcbb20e65bc3d2e9f  arm64=None
  2026.1.1.1-b2-aarch64  pushed=2026-08-18T20:01:48.753231Z  manifest=sha256:d27eede72bbc669e348e9920318589b4bb68e98204a241ec6bfaba04bf4f6699  amd64=None  arm64=sha256:d27eede72bbc669e348e9920318589b4bb68e98204a241ec6bfaba04bf4f6699
  2026.1.1.0-b91  pushed=2026-08-13T15:25:31.439063Z  manifest=sha256:de2e0027864589cf7da61bb2dd5b431f54056977b963a87e123b12f4d64b7cf9  amd64=sha256:efecb2bb82a1b7b95d2ec7e85e23a467055ae87fc7086e3e20981717e3090570  arm64=sha256:79ac9124e6d90d95659c4f455fdd9bd13d987a6714f036681722af0587593e6d
=== https://hub.docker.com/v2/repositories/yugabytedb/yugabyte/tags?name=2.25.&page_size=4&ordering=last_updated ===
count: 9
  2.25.2.0-b359  pushed=2025-05-20T18:34:42.080066Z  manifest=sha256:f1b868431c9f71013f451df8cc63c59886568bb1254e4cb6f98de14d0232a3ad  amd64=sha256:e69ddcd248db7ee5c097d8c955d3657cf7fd9c21743a4589281767487ab5832b  arm64=sha256:3f9cd1b8f520cc78e82ace51174be73343a551148bc6c9a9f821a22f1d2858c5
  2.25.2.0-b359-x86_64  pushed=2025-05-20T18:34:39.248892Z  manifest=sha256:e69ddcd248db7ee5c097d8c955d3657cf7fd9c21743a4589281767487ab5832b  amd64=sha256:e69ddcd248db7ee5c097d8c955d3657cf7fd9c21743a4589281767487ab5832b  arm64=None
  2.25.2.0-b359-aarch64  pushed=2025-05-20T18:32:55.31579Z  manifest=sha256:3f9cd1b8f520cc78e82ace51174be73343a551148bc6c9a9f821a22f1d2858c5  amd64=None  arm64=sha256:3f9cd1b8f520cc78e82ace51174be73343a551148bc6c9a9f821a22f1d2858c5
  2.25.1.0-b381  pushed=2025-03-21T17:12:11.864083Z  manifest=sha256:d56a4f64a8ed12a1a57354a0d68d1974a2aeb519f80c84e3a20519dd38ea8dcd  amd64=sha256:6c836e76cf33be8ce3368bcee9772eff8e1239d49d4e7536aaa9d0f240ee0585  arm64=sha256:e527d56e4a3a78515b5ceb395d304d380dd46c2daa18a04fb2911b049b2bf93d
=== https://hub.docker.com/v2/repositories/yugabytedb/yugabyte/tags?name=2.27.&page_size=4&ordering=last_updated ===
count: 0
=== https://hub.docker.com/v2/repositories/yugabytedb/yugabyte/tags?name=2.29.&page_size=4&ordering=last_updated ===
count: 0
=== https://hub.docker.com/v2/repositories/yugabytedb/yugabyte/tags?name=2.31.&page_size=4&ordering=last_updated ===
count: 0
=== https://hub.docker.com/v2/repositories/yugabytedb/yugabyte/tags?name=2026.2&page_size=4&ordering=last_updated ===
count: 0
=== https://hub.docker.com/v2/repositories/yugabytedb/yugabyte/tags/latest ===
  latest  pushed=2026-08-18T20:03:19.776151Z  manifest=sha256:3926eedf0ff4b99426fbff533128ff16635ed5697a5cf086dde4fafec28e6f75  amd64=sha256:36741c614b0a28498a3447a01b952240905a0ad87c9f68dbcbb20e65bc3d2e9f  arm64=sha256:d27eede72bbc669e348e9920318589b4bb68e98204a241ec6bfaba04bf4f6699
=== https://hub.docker.com/v2/repositories/yugabytedb/yugabyte/tags/2024.2.11.0-b36-aarch64 ===
  2024.2.11.0-b36-aarch64  pushed=2026-08-25T15:59:27.917385Z  manifest=sha256:53c01c56315e245cc901cbc46e93f30fedb1f87833c7b1aa09d20fbeda27a288  amd64=None  arm64=sha256:53c01c56315e245cc901cbc46e93f30fedb1f87833c7b1aa09d20fbeda27a288
=== https://hub.docker.com/v2/repositories/yugabytedb/yugabyte/tags/2025.1.4.0-b103-aarch64 ===
  2025.1.4.0-b103-aarch64  pushed=2026-04-20T15:57:55.830612Z  manifest=sha256:08f1be0d6e074e0286e70067c12933a8cf536876d50fc43c52b1127d447ecdc0  amd64=None  arm64=sha256:08f1be0d6e074e0286e70067c12933a8cf536876d50fc43c52b1127d447ecdc0
=== https://hub.docker.com/v2/repositories/yugabytedb/yugabyte/tags/2025.2.6.0-b111-aarch64 ===
  2025.2.6.0-b111-aarch64  pushed=2026-09-04T22:22:00.73534Z  manifest=sha256:4a1f7a94d3b70c61ecaf5e049521d691b16c9bf204ed01bdb6ab75d5beeb623e  amd64=None  arm64=sha256:4a1f7a94d3b70c61ecaf5e049521d691b16c9bf204ed01bdb6ab75d5beeb623e
=== https://hub.docker.com/v2/repositories/yugabytedb/yugabyte/tags/2026.1.1.1-b2-aarch64 ===
  2026.1.1.1-b2-aarch64  pushed=2026-08-18T20:01:48.753231Z  manifest=sha256:d27eede72bbc669e348e9920318589b4bb68e98204a241ec6bfaba04bf4f6699  amd64=None  arm64=sha256:d27eede72bbc669e348e9920318589b4bb68e98204a241ec6bfaba04bf4f6699
=== https://hub.docker.com/v2/repositories/yugabytedb/yugabyte/tags/2.25.2.0-b359-aarch64 ===
  2.25.2.0-b359-aarch64  pushed=2025-05-20T18:32:55.31579Z  manifest=sha256:3f9cd1b8f520cc78e82ace51174be73343a551148bc6c9a9f821a22f1d2858c5  amd64=None  arm64=sha256:3f9cd1b8f520cc78e82ace51174be73343a551148bc6c9a9f821a22f1d2858c5

**Comment by kriteshranjan** [comment]
Closing this issue based on AI validation. cc: @m-iancu , @VamsidharGunturu
